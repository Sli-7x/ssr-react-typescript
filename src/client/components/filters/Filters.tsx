import * as React from 'react';
import styled from 'styled-components';

export interface IFiltersProps {
  list: any[];
}

const Title = styled.span`
  color: #555;
`;

export class Filters extends React.Component<IFiltersProps, any> {
  render() {
    return (
      <>
        {this.props.list.map((filter: any) => (
          <div key={filter.id}>
            <Title>{filter.title}</Title>
          </div>
        ))}
      </>
    );
  }
}
