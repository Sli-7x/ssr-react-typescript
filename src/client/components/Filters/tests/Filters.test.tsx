import * as React from 'react';
import Filters from '../Filters';
import * as renderer from 'react-test-renderer';

const mock = [
  {
    id: 12,
    title: 'Some random title',
  },
];

describe('Filters.tsx', () => {
  it('should render correctly', () => {
    const filledTree = renderer.create(<Filters list={mock} />).toJSON();
    expect(filledTree).toMatchSnapshot();
  });
});
