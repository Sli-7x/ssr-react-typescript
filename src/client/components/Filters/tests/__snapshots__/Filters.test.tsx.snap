// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters.tsx should render correctly 1`] = `
<div>
  <span
    className="sc-bdVaJa bcQTER"
  >
    Some random title
  </span>
</div>
`;
